test/test_smoke.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright 2025 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "integration_utils.hpp"
+
+class smoke : public CRATempServer {};
+
+TEST_P(smoke, commit) {
+  gpg_error_t rc = assuan_transact(client, "COMMIT", NULL, NULL, NULL, NULL, NULL, NULL);
+  EXPECT_FALSE(rc);
+}
+INSTANTIATE_TEST_SUITE_P( , smoke, testing::Values("empty", "populated"), smoke::PrintParamName);
```

Comment:
This is a pretty bare example of how the fixture can be used.

In the test body, you have a running CRA server (in another thread) and an already connected client context. The fixture directory has already been copied to a temporary directory, so you can modify it as part of the test or test setup. The client and server will be terminated gracefully during the test tear-down.

Note that even though the server process has been started, it hasn't actually loaded any repository metadata, so you can absolutely do that in the test body as long as you haven't invoked any operations that would load the metadata first.